drivers/adc/adc_mcux_adc16.c

```diff
@@ -232,8 +291,33 @@ static int mcux_adc16_init(struct device *dev)
 	ADC_Type *base = config->base;
 	adc16_config_t adc_config;
 
+	LOG_DBG("init adc");
 	ADC16_GetDefaultConfig(&adc_config);
 
+	/*try the fast ADC convert settings*/
+	/* use bus clock*/
+#ifdef CONFIG_ADC16_ENABLE_EDMA
+	if (DT_INST_NODE_HAS_PROP(0, clk_source)) {
```

Comment:
can move move these instance-specific bits into the device config struct?

Reply:
yes, I will change it to config directly. I just find there is a ULTI_AND which can do this tricky